### PR TITLE
Rename generic T and E params

### PR DESCRIPTION
**Pull request by neilpa**
Merged and resolved conflicts building from #2349 since it was outdated

/cc @bencochran

### CODE REVIEW

File: ReactiveCocoa/Swift/Action.swift [hunk 1]
```diff
@@ -202,18 +202,18 @@ public final class CocoaAction: NSObject {
...
 	/// disabled.
 	case NotEnabled
 
 	/// The producer returned from apply() sent the given error.
-	case ProducerError(E)
+	case ProducerError(Error)
 }
 
-public func == <E: Equatable>(lhs: ActionError<E>, rhs: ActionError<E>) -> Bool {
+public func == <Error: Equatable>(lhs: ActionError<Error>, rhs: ActionError<Error>) -> Bool {
```
Comment (NachoSoto) [inline ReactiveCocoa/Swift/Action.swift]: Nit: should there be a space between `==` and `<`?

File: ReactiveCocoa/Swift/Event.swift [hunk 2]
```diff
@@ -10,13 +10,13 @@
 ///
 /// Signals must conform to the grammar:
 /// `Next* (Error | Completed | Interrupted)?`
-public enum Event<T, E: ErrorType> {
+public enum Event<Value, Err: ErrorType> {
```
Comment (NachoSoto) [inline ReactiveCocoa/Swift/Event.swift]: I think I missed this discussion: why `Err` instead of `Error`?

Comment (neilpa) [inline ReactiveCocoa/Swift/Event.swift]: Because it conflicts w/ the `Error` case. See #2360

File: ReactiveCocoa/Swift/Action.swift (hunk 1 above)
Comment (neilpa) [inline ReactiveCocoa/Swift/Action.swift]: Huh? There is already

File: ReactiveCocoa/Swift/Event.swift (hunk 2 above)
Comment (NachoSoto) [inline ReactiveCocoa/Swift/Event.swift]: Ooooh right

File: ReactiveCocoa/Swift/Action.swift (hunk 1 above)
Comment (NachoSoto) [inline ReactiveCocoa/Swift/Action.swift]: It was a rhetorical question :P

Comment (neilpa) [inline ReactiveCocoa/Swift/Action.swift]: Not for normal functions, but for operators I find it really helps with readability.

**Comment by bencochran** [comment]
Thanks for giving this an update

**Comment by neilpa** [comment]
Updated the changelog too, should be ready to go.

**Pull request merged**